source/opt/control_dependence.h

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2021 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_OPT_CONTROL_DEPENDENCE_H_
+#define SOURCE_OPT_CONTROL_DEPENDENCE_H_
+
+#include <algorithm>
+#include <cstdint>
+#include <functional>
+#include <map>
+#include <vector>
+
+#include "source/opt/cfg.h"
+#include "source/opt/dominator_analysis.h"
+
+namespace spvtools {
+namespace opt {
+
+struct ControlDependence {
```

Comment:
That makes sense, but I have a few concerns:

1. A control dependency does not always correspond 1:1 with a CFG edge, so information is lost if we don't store the actual CFG edge.
2. IMO it's best for classification to return something with all the info you need, as we can get all of it from looking at one instruction.

What do you think would be a better organization? Store the actual CFG edge, and expose ClassifyControlDependence?

Reply:
I was thinking this would be a class, where we do not expose any of the data.  Then ClassifyControlDependence then becomes a private member function or it can even get split up into separate member functions.  This give us the freedom to modify the internal representation to whatever we think is best.

Reply:
+1 to what Steven suggested